### PR TITLE
show heroes if a room has deleted name/canonical_alias

### PR DESCRIPTION
**Pull request by ara4n**

### CODE REVIEW

**Comment by ara4n** [comment]
@erikjohnston ptal again

File: synapse/handlers/sync.py [hunk 1]
```diff
@@ -562,8 +562,21 @@ def compute_summary(self, room_id, sync_config, batch, state, now_token):
...
 
-        if name_id or canonical_alias_id:
-            defer.returnValue(summary)
+        # if the room has a (non-deleted) name or canonical_alias, we can skip
+        # calculating heroes.  we assume that if the event has contents, it'll
+        # be a valid name or canonical_alias - i.e. we're checking that they
+        # haven't been deleted.
+        if name_id:
+            name = yield self.store.get_event(name_id, allow_none=False)
+            if name.content:
```
Comment (erikjohnston) [inline synapse/handlers/sync.py]: You also need to check that the returned event is not None

File: synapse/handlers/sync.py [hunk 2]
```diff
@@ -562,8 +562,21 @@ def compute_summary(self, room_id, sync_config, batch, state, now_token):
             details.get(Membership.INVITE, empty_ms).count
         )
 
-        if name_id or canonical_alias_id:
-            defer.returnValue(summary)
+        # if the room has a (non-deleted) name or canonical_alias, we can skip
```
Comment (erikjohnston) [inline synapse/handlers/sync.py]: What do you mean by non-deleted? You can't delete state.

Comment (ara4n) [reply synapse/handlers/sync.py]: well, people 'delete' it by blatting {} over it.

File: synapse/handlers/sync.py (hunk 1 above)
Comment (ara4n) [reply synapse/handlers/sync.py]: oops, thanks

File: synapse/handlers/sync.py (hunk 2 above)
Comment (ara4n) [reply synapse/handlers/sync.py]: clarified

**Comment by ara4n** [comment]
@erikjohnston ptal again

**Pull request merged**